runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -1127,13 +1220,74 @@ TR_SharedCacheRelocationRuntime::storeAOTHeader(TR_FrontEnd *fe, J9VMThread *cur
       }
    }
 
+
+#if defined(JITSERVER_SUPPORT)
+void *
+TR_SharedCacheRelocationRuntime::isROMClassInSharedCaches(UDATA romClassValue)
+   {
+   j9thread_monitor_enter(javaVM()->sharedClassConfig->configMonitor);
+   J9SharedClassCacheDescriptor *currentCacheDescriptor = javaVM()->sharedClassConfig->cacheDescriptorList;
+   bool matchFound = false;
+
+   while (!matchFound && currentCacheDescriptor)
```

Comment:
Conceptually, `currentCacheDescriptor` cannot be NULL because the list is circular. Maybe only if there is no SCC at all.

Reply:
Agree, will keep the check just in case `cacheDescriptorList` is NULL.